lib/cli_common/cli_common/phabricator.py

```
@@ -4,44 +4,216 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 import functools
+import logging
+from urllib.parse import urlparse
 
 import requests
 
-PHABRICATOR_API_URL_TEMPLATE = 'https://phabricator.services.mozilla.com/api/{}'
 HGMO_JSON_REV_URL_TEMPLATE = 'https://hg.mozilla.org/mozilla-central/json-rev/{}'
+MOZILLA_PHABRICATOR_PROD = 'https://phabricator.services.mozilla.com/api/'
+
+
+logger = logging.getLogger(__name__)
 
 
 @functools.lru_cache(maxsize=2048)
-def _revision_exists_on_central(revision):
+def revision_exists_on_central(revision):
     url = HGMO_JSON_REV_URL_TEMPLATE.format(revision)
-    return requests.get(url).ok
+    try:
```

Maybe it's better to just throw an exception when we can't hit hgweb?